src/java/org/apache/cassandra/config/BitRate.java

```diff
@@ -0,0 +1,331 @@
+     */
+    public BitRateUnit getUnit()
+    {
+        return unit;
+    }
+
+    /**
+     * Returns the bit rate in bits per seconds
+     *
+     * @return the bit rate in bits per seconds
+     */
+    public long toBitsPerSecond()
+    {
+        return unit.toBitsPerSecond(quantity);
+    }
+
+    /**
+     * Returns the bit rate in bits per seconds as an {@code int}
+     *
+     * @return the bit rate in bits per secondss or {@code Integer.MAX_VALUE} if the rate is too large.
+     */
+    public int toBitsPerSecondAsInt()
+    {
+        return Ints.saturatedCast(toBitsPerSecond());
+    }
+
+    /**
+     * Returns the bit rate in kilobits per seconds
+     *
+     * @return the bit rate in kilobits per seconds
+     */
+    public long toKilobitsPerSecond()
+    {
+        return unit.toKilobitsPerSecond(quantity);
+    }
+
+    /**
+     * Returns the bit rate in kilobits per seconds as an {@code int}
+     *
+     * @return the bit rate in kilobits per seconds or {@code Integer.MAX_VALUE} if the number of kilobits is too large.
+     */
+    public int toKilobitsPerSecondAsInt()
+    {
+        return Ints.saturatedCast(toKilobitsPerSecond());
+    }
+
+    /**
+     * Returns the bit rate in megabits per seconds
+     *
+     * @return the bit rate in megabits per seconds
+     */
+    public long toMegabitsPerSecond()
+    {
+        return unit.toMegabitsPerSeconds(quantity);
+    }
+
+    /**
+     * Returns the bit rate in megabits per seconds as an {@code int}
+     *
+     * @return the bit rate in megabits per seconds or {@code Integer.MAX_VALUE} if the number of megabits is too large.
+     */
+    public int toMegabitsPerSecondAsInt()
+    {
+        return Ints.saturatedCast(toMegabitsPerSecond());
+    }
+
+    @Override
+    public int hashCode()
+    {
+        return Objects.hash(unit.toKilobitsPerSecond(quantity));
+    }
+
+    @Override
+    public boolean equals(Object obj)
+    {
+        if (this == obj)
+            return true;
+
+        if (!(obj instanceof BitRate))
+            return false;
+
+        BitRate other = (BitRate) obj;
+        if (unit == other.unit)
+            return quantity == other.quantity;
+
+        // Due to overflows we can only guarantee that the 2 bit rates are equal if we get the same results
+        // doing the conversion in both directions.
+        return unit.convert(other.quantity, other.unit) == quantity && other.unit.convert(quantity, unit) == other.quantity;
+    }
+
+    @Override
+    public String toString()
+    {
+        return quantity + unit.symbol;
+    }
+
+    public String quantityToString()
+    {
+        return String.valueOf(quantity);
+    }
+
+    public enum BitRateUnit
+    {
+        BITS_PER_SECOND("b/s")
+        {
+            public long toBitsPerSecond(long d)
+            {
+                return d;
+            }
+
+            public long toKilobitsPerSecond(long d)
+            {
+                return d / 1000;
```

Comment:
not consistent with `DataStorage`; we use different units in different raveled classes?

1024 = kibi / mebi
1000 = kilo / mega

see https://www.majordifferences.com/2018/03/differences-between-megabyte-and.html

Reply:
Valid point, not intentional. I will check which one we want and align them

Reply:
mebi, both will be corrected in DataStorage and BitRate